### ISSUE TITLE
ngx.header can also be run in set_by_lua block

### ISSUE THREAD

**Issue by advancingyang** [openresty/lua-nginx-module#1485]
As the README states(https://github.com/openresty/lua-nginx-module#ngxheaderheader), ngx.header is not supposed to be able to run in set_by_lua? 
`context: rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua*, log_by_lua*`

I test locally that do `ngx.header['CUSTOM_HEADER'] = 'WHATEVER'` in set_by_lua, seem I can see the headers in the HTTP response. Is that something not aligned with README?

**Comment by agentzh** [comment]
@advancingyang Yeah, we can add that to the docs. The docs may miss things since it's written by humans ;) Pull requests welcome! Thanks.
